core/dataset_manager.py

```diff
@@ -152,69 +160,68 @@ def _create_dataset_mappings(self, mapping_filepath):
-        under the provided name as the key
+        Post the ED Directory on blockchain with the given key.
+
+        The ED Directory is a JSON dictionary whose keys represent the dataset
+        folders in the directory and whose values represent the corresponding
+        datasets. Assume only one dataset file per folder. 
 
-        IMPORTANT: NOT FINISHED DEBUGGING, DO NOT USE
+        See _generate_ed_directory docstring for an example of what an ED
+        Directory looks like.
         """
-        filepath = self._raw_filepath
-        self.check_key_length(name)
-        value = {}
-        folders = []
-        for file in os.listdir(filepath):
-            if os.path.isdir(os.path.join(os.path.abspath(filepath), file)):
-                folders.append(file)
-        for folder in folders:
-            folder_dict = {}
-            folder_path = os.path.join(os.path.abspath(filepath), folder)
-            files = os.listdir(folder_path)
-            for file in files:
-                if file[:2] == 'md':
-                    file_path = os.path.join(folder_path, file)
-                    metadata = pd.read_csv(file_path)
-                    folder_dict['md'] = metadata.to_json()
-                else:
-                    file_path = os.path.join(folder_path, file)
-                    dataset = pd.read_csv(file_path)
-                    sample = dataset.sample(frac=0.1)
-                    folder_dict['ds'] = sample.to_json()
-            if 'md' not in folder_dict:
-                raise NoMetadataFoundError(folder)
-            value[folder] = folder_dict
-        receipt = setter(client=self._client, key=name, value=value, port=self._port)
-
-    def post_dataset(self, name):
-        """
-        Post samples of datasets on blockchain with automatically generated
-        metadata under provided name as the key
-
-        IMPORTANT: NOT FINISHED DEBUGGING, DO NOT USE
-        """
-        filepath = self._raw_filepath
-        self.check_key_length(name)
-        value = {}
-        folders = []
-        for file in os.listdir(filepath):
-            if os.path.isdir(os.path.join(os.path.abspath(filepath), file)):
-                folders.append(file)
-        for folder in folders:
-            folder_dict = {}
-            folder_path = os.path.join(os.path.abspath(filepath), folder)
-            file = list(os.listdir(folder_path))[0]
-            file_path = os.path.join(folder_path, file)
-            dataset = pd.read_csv(file_path)
-            md = pd.DataFrame(dataset.describe())
-            sample = dataset.sample(frac=0.1)
-            folder_dict['ds'] = sample.to_json()
-            folder_dict['md'] = md.to_json()
-            value[folder] = folder_dict
-        receipt = setter(client=self._client, key=name, value=value, port=self._port)
+        assert len(key) <= 30, \
+            "Keys for datasets can only be at most 30 characters long."
```

Comment:
Why is this? I don't think this is a restriction that we have for the blockchain. Is this just an arbitrary length? I'm also not sure what the "dataset key" is.

Reply:
This was a restriction you gave me way back in the first iteration of Dataset Manager. If it's not relevant anymore, then I can remove it.

Also yeah dataset key is a typo, I meant to say data provider key

Reply:
Technically not relevant since Ethereum -> Tendermint but we should leave it in so that we don't break anything.